packages/web/src/app/doc-pages/brand/icon-doc/icon-preview/icon-preview-details/icon-preview-details.component.ts

```diff
@@ -14,7 +14,39 @@ import { CopyComponent } from 'src/app/shared/copy/copy.component';
   styleUrl: './icon-preview-details.component.scss',
 })
 export class IconPreviewDetailsComponent {
+  private elementRef = inject(ElementRef);
+
   selectedIcon = model.required<Icon | null>();
+  theme = input.required<Theme>();
+
+  downloadImage = async (format: 'svg' | 'png') => {
+    const svgElement = this.getIconElement();
+    if (!svgElement) {
+      throw new Error('No SVG element found');
+    }
+
+    const a = document.createElement('a');
+    a.href = await (format === 'svg'
+      ? createSvgBlobFromElement(svgElement, {
+          tokens: 'icon',
+          theme: this.theme(),
+        })
+      : createPngBlob(svgElement, {
+          tokens: 'icon',
+          theme: this.theme(),
+        }));
+    a.download = this.imageFileName(format);
+    a.click();
+  };
+
```

Comment:
Does this element need to be removed once clicked?

Reply:
Good question. It's never attached to the DOM, so I'm actually not quite sure

Reply:
According to this response here, it seems we don't need to remove the element since it's never attached to the DOM. 
https://stackoverflow.com/a/62536956